docs/architecture/deep-dives/passkeys/credentials.md

```diff
@@ -0,0 +1,89 @@
+---
+sidebar_position: 2
+---
+
+# Credentials
+
+This page provides an in-depth look at the FIDO2 credential, which is the technical term used to
+refer to a passkey in the context of the FIDO2 standard. On this page they will be referred to as
+simply "credentials".
+
+## What is a Credential?
+
+Technically the term "credential" refers to the public/private key pair that is generated by a FIDO2
+authenticator. The private key is protected by the authenticator, and the public key is stored
+server-side by the Relaying Party (RP). The authenticator uses the private key to sign challenges
+from a RP, and the RP uses the public key to verify the signature.
+
+On this page the term "credential" will be used to refer the data structure that is protected by the
+authenticator, including the private key, and any metadata that is associated with it.
+
+## Data structure
+
+### Required fields
+
+Credentials are relatively simple data structures and only require the following fields:
+
+- `credentialId` - Binary data sequence that uniquely identifies the credential. It is either a
+  random value, or an encrypted version of the credential (see [Storage Modes](#storage-modality)).
+- `rpId` - The identifier of the RP for which the credential was created. By default assigned the
+  effective domain of the applications origin (ex: bitwarden.com).
+- `privateKey` - Private key material, created during the creation operation.
+
+### Notable optional fields
+
+- `userHandle` - Binary data sequence that uniquely identifies the user account. This data is opaque
+  for anyone other than the RP. It is used to associate a credential with a specific user account.
+- `userName` - A human-readable name for the user account. This can be used to display the user
+  account in the authenticator's user interface (if supported).
+
+## Storage Modality
+
+### Client-Side Credentials
+
+The credential is stored in persistent storage embedded in the authenticator, client or client
+device. This is rapidly becoming a common storage modality, as it is the only way to support synced
```

Comment:
issue: This should say `mode` and not `modality` (applies in more places on this page)

Modality is "the quality or state of being modal", while mode is "a particular form or variety of something". 

```suggestion
device. This is rapidly becoming a common storage mode, as it is the only way to support synced
```